streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -314,9 +341,29 @@ public synchronized final TopologyBuilder addSource(String name, String... topic
      * @return this builder instance so methods can be chained together; never null
      */
     public synchronized final TopologyBuilder addSource(String name, Pattern topicPattern) {
-        return addSource(name, (Deserializer) null, (Deserializer) null, topicPattern);
+        return addSource(name, null, (Deserializer) null, (Deserializer) null, topicPattern);
+    }
+
+    /**
+     * Add a new source that consumes from topics matching the given pattern
+     * and forward the records to child processor and/or sink nodes.
+     * The source will use the {@link org.apache.kafka.streams.StreamsConfig#KEY_SERDE_CLASS_CONFIG default key deserializer} and
+     * {@link org.apache.kafka.streams.StreamsConfig#VALUE_SERDE_CLASS_CONFIG default value deserializer} specified in the
+     * {@link org.apache.kafka.streams.StreamsConfig stream configuration}.
+     *
+     * @param name the unique name of the source used to reference this node when
+     * @param offsetReset the offset reset value for this source acceptable values earliest or latest.  For this parameter
```

Comment:
If I understand your question correctly, if we don't set the `AUTO_OFFSET_RESET_CONFIG` to `none`  then the  `NoOffsetForPartitionException` is not thrown, and we don't get the chance to do the see/position API calls.